include/engine/api/route_api.hpp

```diff
@@ -439,7 +439,7 @@ class RouteAPI : public BaseAPI
     {
         // AnnotationsType uses bit flags, & operator checks if a property is set
         flatbuffers::Offset<flatbuffers::Vector<float>> speed;
-        if (parameters.annotations_type & RouteParameters::AnnotationsType::Speed)
+        if (requested_annotations & RouteParameters::AnnotationsType::Speed)
```

Comment:
We can leave it "as is" and NodeJs tests will still pass, but this seems to fix the problem for libosrm(when `annotations = true` was used).

See:
https://github.com/Project-OSRM/osrm-backend/pull/6415/files#r998472037